java-checks/src/main/java/org/sonar/java/checks/unused/UnusedLocalVariableCheck.java

```diff
@@ -90,11 +90,12 @@ public void leaveNode(Tree tree) {
   }
 
   /**
-   * Before Java 22 it was not possible to remove the variable in a foreach statement even it is unused.
+   * Before Java 22 it was not possible to remove the variable in a foreach statement or try with resources even it is unused.
    * For instance in {@code for (String element : list) {}}, it is only since Java 22 that it can be rewritten {@code for (var _ : list) {}}.
    */
-  private static boolean isMandatoryForeachVariable(JavaVersion javaVersion, VariableTree variable) {
-    return variable.parent() instanceof ForEachStatement && !javaVersion.isJava22Compatible();
+  private boolean isMandatory(VariableTree variable) {
```

Comment:
```suggestion
  private boolean canBeReplaced(VariableTree variable) {
  return context.getJavaVersion().isJava22Compatible()
      || (!isForeachVariable(variable) && !isTryResource(variable));
  }
```
I find that the negation on mandatory make it a bit complex to read. We can avoid it by taking the negation of isMandtory with canBeReplaced